### PR TITLE
Templates.Blazor.Tests: skip Playwright based tests when Playwright is not available.

### PR DESCRIPTION
**Pull request by tmds**
Playwright is not available on some distros, like RHEL, CentOS, and Fedora.
This causes the `Templates.Blazor.Tests` test suite to fail when ran on those distros.

Rather than failing, these tests should be skipped.

This uses the existing `IsPlaywrightAvailable` project property.
When it is `false`, the tests are skipped by making the `BrowserManager` return the browsers as as unavailable/disabled.

This enables running the test suite on a platform where Playwright is not available by appending `/p:IsPlaywrightAvailable=false` to the build command.

@dougbu ptal.

cc @omajid

### CODE REVIEW

**Comment by dougbu** [comment]
@tmds, please mention @dotnet/aspnet-blazor-eng when updating Blazor-related bits.

Somewhat separately, I'm not sure about this approach. Suggest we should instead use our normal [`ConditionalTheory`](https://github.com/dotnet/aspnetcore/blob/main/src/Testing/src/xunit/ConditionalTheoryAttribute.cs) and [`ConditionalFact`](https://github.com/dotnet/aspnetcore/blob/main/src/Testing/src/xunit/ConditionalFactAttribute.cs) attributes. Add a new [`ITestCondition`](https://github.com/dotnet/aspnetcore/blob/main/src/Testing/src/xunit/ITestCondition.cs) that checks whether Playwright is supported on the platform. Search the repo to see how that all works.

**Comment by tmds** [comment]
> Somewhat separately, I'm not sure about this approach. Suggest we should instead use our normal [ConditionalTheory](https://github.com/dotnet/aspnetcore/blob/main/src/Testing/src/xunit/ConditionalTheoryAttribute.cs) and [ConditionalFact](https://github.com/dotnet/aspnetcore/blob/main/src/Testing/src/xunit/ConditionalFactAttribute.cs) attributes. Add a new [ITestCondition](https://github.com/dotnet/aspnetcore/blob/main/src/Testing/src/xunit/ITestCondition.cs) that checks whether Playwright is supported on the platform. Search the repo to see how that all works.

There are several ways to do this.

I opted for this because the `BrowserManager` already has the concept of unavailable/disabled browsers.
The tests themselves are written Playwright-agnostic, and have a code-path to handle the 'unavailable' case.

I'll adjust based on feedback.

**Comment by tmds** [comment]
@dotnet/aspnet-blazor-eng ptal.

If it's not desired to directly do this based on `IsPlaywrightAvailable`, I can add another property on top, like`DisablePlaywright`, which defaults to `false` and can be set on the command-line to `true`.

Or if you prefer to use an `Attribute` on each test: what condition should it use to skip?

**Comment by tmds** [comment]
@dotnet/aspnet-blazor-eng ptal.

Without these changes, the aspnetcore tests fail on our internal CI which tests against RHEL and Fedora distros.

**Review by javiercn — APPROVED** [review]

File: src/Shared/BrowserTesting/src/Microsoft.AspNetCore.BrowserTesting.csproj [hunk 1]
```diff
@@ -6,6 +6,7 @@
     <IsShippingPackage>false</IsShippingPackage>
     <AddPublicApiAnalyzers>false</AddPublicApiAnalyzers>
     <IsTestAssetProject>true</IsTestAssetProject>
+    <DefineConstants Condition="'$(IsPlaywrightAvailable)' != 'true'">$(DefineConstants);DISABLE_PLAYWRIGHT</DefineConstants>
```
Comment (javiercn) [inline src/Shared/BrowserTesting/src/Microsoft.AspNetCore.BrowserTesting.csproj:9]: Doesn't this disable our tests by default? Where does `IsPlaywrightAvailable` come from?

Shouldn't this be
```suggestion
    <DefineConstants Condition="'$(IsPlaywrightAvailable)' == 'false'">$(DefineConstants);DISABLE_PLAYWRIGHT</DefineConstants>
```

Comment (tmds) [reply src/Shared/BrowserTesting/src/Microsoft.AspNetCore.BrowserTesting.csproj:9]: It comes from the top-level `Directory.Build.props`:

https://github.com/dotnet/aspnetcore/blob/afd72717c019eb2b8f4e56ad39d1898a6d333fbc/Directory.Build.props#L180-L181

Comment (tmds) [reply src/Shared/BrowserTesting/src/Microsoft.AspNetCore.BrowserTesting.csproj]: Though it was functional, I've committed your suggestion as it is safer if something would happen to the top-level `IsPlaywrightAvailable`.

Comment (javiercn) [reply src/Shared/BrowserTesting/src/Microsoft.AspNetCore.BrowserTesting.csproj]: Thanks!

**Comment by tmds** [comment]
The CI failures look unrelated.

Is this good to merge?

**Pull request merged**